src/osd/PrimaryLogPG.cc

```diff
@@ -348,21 +348,28 @@ void PrimaryLogPG::on_local_recover(
     set<snapid_t> snaps;
     dout(20) << " snapset " << recovery_info.ss
 	     << " legacy_snaps " << recovery_info.oi.legacy_snaps << dendl;
+    bool error = false;
     if (recovery_info.ss.is_legacy() ||
 	recovery_info.ss.seq == 0 /* jewel osd doesn't populate this */) {
       assert(recovery_info.oi.legacy_snaps.size());
       snaps.insert(recovery_info.oi.legacy_snaps.begin(),
 		   recovery_info.oi.legacy_snaps.end());
     } else {
       auto p = recovery_info.ss.clone_snaps.find(hoid.snap);
-      assert(p != recovery_info.ss.clone_snaps.end());  // hmm, should we warn?
-      snaps.insert(p->second.begin(), p->second.end());
-    }
-    dout(20) << " snaps " << snaps << dendl;
-    snap_mapper.add_oid(
-      recovery_info.soid,
-      snaps,
-      &_t);
+      if (p != recovery_info.ss.clone_snaps.end()) {
+        snaps.insert(p->second.begin(), p->second.end());
+      } else {
+        derr << __func__ << " " << hoid << " had no clone_snaps" << dendl;
+        error = true;
```

Comment:
I don't thing you need the bool error.  Isn't there just an extra indentation, but make the same exact change as in master?

Reply:
@dzafman No, IIRC you advised me to resolve the conflict this way.

Reply:
because master doesn't have this `if (recovery_info.ss.is_legacy())`

Reply:
@smithfarm Yup, you/I was right.